### PR TITLE
remove ConnectMessage,  use events instead

### PR DESCRIPTION
**Pull request by paulpach**

### CODE REVIEW

**Comment by paulpach** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- Assets/Mirror/Tests/Play/AsyncUtil.cs  3
         

Clones added
============
- Assets/Mirror/Tests/Play/NetworkManagerTest.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/paulpach/MirrorNG/pullRequest?prid=5221840&bid=17121201)

**Comment by paulpach** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- Assets/Mirror/Tests/Play/AsyncUtil.cs  3
         

Clones added
============
- Assets/Mirror/Tests/Play/NetworkManagerTest.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/paulpach/MirrorNG/pullRequest?prid=5221840&bid=17121201)

**Comment by paulpach** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- Assets/Mirror/Tests/Play/AsyncUtil.cs  3
         

Clones added
============
- Assets/Mirror/Tests/Play/NetworkManagerTest.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/paulpach/MirrorNG/pullRequest?prid=5221840&bid=17121201)

**Comment by paulpach** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           

Complexity increasing per file
==============================
- Assets/Mirror/Tests/Play/AsyncUtil.cs  3
         

Clones added
============
- Assets/Mirror/Tests/Play/NetworkManagerTest.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/paulpach/MirrorNG/pullRequest?prid=5221840&bid=17121201)

**Comment by paulpach** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           

Complexity increasing per file
==============================
- Assets/Mirror/Tests/Play/AsyncUtil.cs  3
         

Clones added
============
- Assets/Mirror/Tests/Play/NetworkManagerTest.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/paulpach/MirrorNG/pullRequest?prid=5221840&bid=17121201)

File: Assets/Mirror/Runtime/NetworkServer.cs [hunk 1]
```diff
@@ -207,8 +207,12 @@ internal void SetLocalConnection(NetworkClient client, ULocalConnectionToClient
                 return;
             }
 
+            conn.SetHandlers(handlers);
```
Comment (Lymdun) [inline Assets/Mirror/Runtime/NetworkServer.cs]: May I ask why did you added this line?

**Comment by paulpach** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           

Complexity increasing per file
==============================
- Assets/Mirror/Tests/Play/AsyncUtil.cs  3
         

Clones added
============
- Assets/Mirror/Tests/Play/NetworkManagerTest.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/paulpach/MirrorNG/pullRequest?prid=5221840&bid=17121201)

File: Assets/Mirror/Runtime/NetworkManager.cs [hunk 2]
```diff
@@ -489,7 +492,6 @@ void StartHostClient()
             }
 
             server.ActivateHostScene();
-            RegisterClientMessages();
 
             // ConnectLocalServer needs to be called AFTER RegisterClientMessages
```
Comment (Lymdun) [inline Assets/Mirror/Runtime/NetworkManager.cs]: What about this?

**Comment by Lymdun** [comment]
Nice job 👍

File: Assets/Mirror/Runtime/NetworkManager.cs (hunk 2 above)
Comment (paulpach) [reply Assets/Mirror/Runtime/NetworkManager.cs]: got replaced by the setHandlers.   This whole thing will go away soon once I move RegisterHandlers to the connection.
This is so convoluted right now.

Comment (Lymdun) [reply Assets/Mirror/Runtime/NetworkManager.cs]: So, it is not an issue anymore if it is called before ConnectLocalServer?

Comment (paulpach) [reply Assets/Mirror/Runtime/NetworkManager.cs]: yeap,  no issue anymore

**Pull request merged**